schematics/table/files/__path__/__name@dasherize@if-flat__/__name@dasherize__.component.ts

```diff
@@ -0,0 +1,131 @@
+import { Component, OnInit, ViewChild, <% if(!!viewEncapsulation) { %>, ViewEncapsulation<% }%><% if(changeDetection !== 'Default') { %>, ChangeDetectionStrategy<% }%> } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { map } from 'rxjs/operators/map';
+import { MatPaginator, MatSort } from '@angular/material';
+import { DataSource } from '@angular/cdk/collections';
+import { of } from 'rxjs/observable/of';
+import { merge } from 'rxjs/observable/merge';
+
+/** Todo: Replace this with your own data model type */
+export interface DataItem {
+  name: string;
+  id: number;
+}
+
+function compare(a, b, isAsc) {
+  return (a < b ? -1 : 1) * (isAsc ? 1 : -1);
+}
+
+export class ExampleDataSource extends DataSource<any> {
+  data: DataItem[] = [
+    {id: 1, name: 'Hydrogen'},
+    {id: 2, name: 'Helium'},
+    {id: 3, name: 'Lithium'},
+    {id: 4, name: 'Beryllium'},
+    {id: 5, name: 'Boron'},
+    {id: 6, name: 'Carbon'},
+    {id: 7, name: 'Nitrogen'},
+    {id: 8, name: 'Oxygen'},
+    {id: 9, name: 'Fluorine'},
+    {id: 10, name: 'Neon'},
+    {id: 11, name: 'Sodium'},
+    {id: 12, name: 'Magnesium'},
+    {id: 13, name: 'Aluminum'},
+    {id: 14, name: 'Silicon'},
+    {id: 15, name: 'Phosphorus'},
+    {id: 16, name: 'Sulfur'},
+    {id: 17, name: 'Chlorine'},
+    {id: 18, name: 'Argon'},
+    {id: 19, name: 'Potassium'},
+    {id: 20, name: 'Calcium'},
+  ];
+
+  constructor(private _paginator: MatPaginator, private _sort: MatSort) {
```

Comment:
I've been thinking about this, and I'm thinking it would be good to decouple the data source and the sorter/paginator. This would look something like the data source having its own sort and pagination state, and then using the outputs on the directives to update the data source state. 

My main reason for wanting to do this is to not make users think they *must* use those components.

Reply:
Ya, I can see that but it wouldn't be near as clean then. What about a comment instead? Best way I could think of would be to:

```
this.dataSource = new DataSource({
  pageIndex: this.paginator.pageIndex,
  pageSize: this.paginator.pageSize
}, {
  active: this.sort.active,
  direction: this.sort.direction
});
```